app/routes.py

```diff
@@ -1 +1,212 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, jsonify, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+import datetime
+from dotenv import load_dotenv
+import requests
+import os
+
+load_dotenv()
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# validation helper function
+
+def validate_item(model, item_id):
+    try:
+        item_id = int(item_id)
+    except ValueError:
+        abort(make_response({"msg": f"{model.__name__} {item_id} invalid"}, 400))
+    
+    item = model.query.get(item_id)
+    
+    if not item:
+        abort(make_response({"msg": f"{model.__name__} {item_id} not found"}, 404))
+    
+    return item
+
+# Task model routes
+
+@task_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+    except KeyError:
+        return {
+            "details": "Invalid data"
+        }, 400
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+@task_bp.route("", methods=["GET"])
+def get_tasks():
+    response = []
+
+    title_query = request.args.get("title")
+    sort_query = request.args.get("sort")
+
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+    elif sort_query == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    
+    #refactor to do list comprehension
+    for task in tasks: 
+        response.append(task.to_dict())
+
+    
+    print(tasks)
```

Comment:
Be sure to remove unused code, print tests, and any comments that shouldn't be published before finalizing a project. It's good practice to keep a clean main branch that's production ready. This helps with the readability and maintainability of a project. 🧹😌